tests/misc/CachingUtilitiesTest.py

```diff
@@ -0,0 +1,53 @@
+import os
+import unittest
+
+from pyprint.NullPrinter import NullPrinter
+
+from coalib.misc.CachingUtilities import (
+    get_data_path, pickle_load, pickle_dump, delete_cache_files)
+from coalib.misc.Caching import FileCache
+from coalib.output.printers.LogPrinter import LogPrinter
+
+
+class CachingUtilitiesTest(unittest.TestCase):
+
+    def setUp(self):
+        self.log_printer = LogPrinter(NullPrinter())
+
+    def test_pickling(self):
+        test_data = {"answer": 42}
+
+        pickle_dump(self.log_printer, "test_file", test_data)
+        self.assertEqual(pickle_load(self.log_printer, "test_file"), test_data)
+        os.remove(get_data_path(self.log_printer, "test_file"))
+
+        self.assertEqual(pickle_load(
+            self.log_printer, "nonexistant_file"), None)
+        self.assertEqual(pickle_load(
+            self.log_printer, "nonexistant_file", fallback=42), 42)
```

Comment:
while we're at it: those are seperate cases for the same function, rather put them in seperate functions to make sure they stay separate